api/room/schema_query.py

```diff
@@ -264,3 +276,27 @@ def resolve_analytics_for_daily_room_events(self, info,
             )
 
         return all_days_events
+
+    @Auth.user_roles('Admin')
+    def resolve_analytics_for_most_booked_rooms(self, info, start_date, end_date=None):  # noqa: E501
+        query = Room.get_query(info)
+        active_rooms = query.filter(RoomModel.state == "active")
+        most_booked = RoomAnalytics.get_most_booked_rooms(
```

Comment:
Apply the indexing for getting the top ten rooms at this level, same to least booked rooms

Reply:
You are also supposed to refactor the rest endpoint